### PR TITLE
Upgrade to StatsBase v0.34

### PR DESCRIPTION
**Pull request by AsafManela**
1. StatsBase.jl v0.34 removed some internal array types used by MLBase. This change adds a dependency to NumericalTypeAliases.jl that provides these type aliases.

https://github.com/JuliaStats/StatsBase.jl/pull/840

2. Also adding a CompatHelper action

3. Bumps MLBase version so it can be tagged

### CODE REVIEW

**Comment by AsafManela** [comment]
Hi @ararslan , would you mind taking a look and merging? I need this for a couple of downstream packages.

**Comment by ararslan** [comment]
It seems worthwhile to me to replace the uses of those type aliases with their definitions in the same manner as https://github.com/JuliaStats/StatsBase.jl/pull/840 rather than adding a new dependency on a package that defines them.

**Comment by AsafManela** [comment]
I guess I missed your sarcasm here https://github.com/JuliaStats/StatsBase.jl/pull/840#pullrequestreview-1190948271 
I'll try to do that.

**Comment by ararslan** [comment]
Ah, my apologies, the :trollface: in my comment you linked was intended to convey that it wasn't a serious suggestion. I can see how that could be unclear so I hope my comment hasn't been accidentally misguiding people...

**Comment by AsafManela** [comment]
No worries. I made those changes. Hopefully good to go.

File: Project.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 [compat]
 IterTools = "1"
 Reexport = "1"
-StatsBase = "0.33"
+StatsBase = "0.34"
```
Comment (ararslan) [inline Project.toml:14]:
```suggestion
StatsBase = "0.33, 0.34"
```
The change made is compatible with 0.33 as well

**Review by ararslan — APPROVED** [review]
LGTM with a minor tweak to compatibility bounds

**Comment by ararslan** [comment]
Registering here: https://github.com/JuliaRegistries/General/pull/89850

Thanks!

**Comment by AsafManela** [comment]
Thanks for the quick turnaround!

**Pull request merged**